rclpy/rclpy/executors.py

```diff
@@ -844,8 +895,9 @@ def _spin_once_impl(
             pass
         else:
             handler()
-            if handler.exception() is not None:
-                raise handler.exception()
+            exception = handler.exception()
+            if exception is not None:
+                raise exception
```

Comment:
I think this was kind of bug to cause the internal state mismatch? when `handler.exception()` is called `_exception_fetched` is already true, that means we should keep that fetched exception like this. just checking.

Reply:
Yes I believe it could cause a mismatch in case somehow some other function got called in between so storing the exception should avoid it.